src/main/java/org/broadinstitute/hellbender/utils/hdf5/HDF5File.java

```diff
@@ -277,6 +280,54 @@ public double readDouble(final String fullPath) {
         return result;
     }
 
+    /**
+     * Reads a matrix from the underlying HDF5 file.
+     * @param fullPath the full path to the matrix data-set within the HDF5 file.
+     * @return never {@code null}.
+     * @throws GATKException if the matrix does not exist or any other HDF5 level error occurred.
+     */
+    public RealMatrix readMatrix(final String fullPath) {
+        final double[][] values = readDoubleMatrix(fullPath);
+        return new Array2DRowRealMatrix(values, false);
+    }
+
+    /**
+     * Reads a matrix from the underlying HDF5 file and checks its dimensions.
+     * @param fullPath the target data-set full path within the HDF5 file.
+     * @param expectedRowCount the expected number of rows.
+     * @param expectedColumnCount the expected number of columns.
+     * @return GATKException if the result matrix dimensions do not match the expectations or
+     *  any other cause as described in {@link #readMatrix(String)}.
+     */
+    public RealMatrix readMatrixAndCheckDimensions(final String fullPath, final int expectedRowCount, final int expectedColumnCount) {
+        return readMatrixAndCheckDimensions(fullPath, r -> r == expectedRowCount, c -> c == expectedColumnCount);
+    }
+
+    /**
+     * Reads a matrix from the underlying HDF5 file and checks its dimensions.
+     * @param fullPath the target data-set full path within the HDF5 file.
+     * @param expectedRowCount a predicate that returns true iff its argument is an expected number of rows.
+     * @param expectedColumnCount a predicate that returns true iff its argument is an expected number of columns.
+     * @return GATKException if the result matrix dimensions do not match the expectations or
+     *  any other cause as described in {@link #readMatrix(String)}.
+     */
+    public RealMatrix readMatrixAndCheckDimensions(final String fullPath, final IntPredicate expectedRowCount, final IntPredicate expectedColumnCount) {
+        final RealMatrix result = readMatrix(fullPath);
+        if (expectedRowCount.test(result.getRowDimension())
+                && expectedColumnCount.test(result.getColumnDimension())) {
+            return result;
+        }
+        final RealMatrix transpose = result.transpose();
+        if (!expectedRowCount.test(transpose.getRowDimension())) {
```

Comment:
Why does this need to be an `IntPredicate` instead of an `int`?  Is this ever passes a predicate that is not checking for equality with an expected value?